pkg/volume/util/nestedpendingoperations/nestedpendingoperations_test.go

```diff
@@ -264,6 +265,41 @@ func Test_NewGoRoutineMap_Negative_SecondOpBeforeFirstCompletes(t *testing.T) {
 	}
 }
 
+func Test_NewGoRoutineMap_Negative_SecondThirdOpWithDifferentNames(t *testing.T) {
+	// Arrange
+	grm := NewNestedPendingOperations(true /* exponentialBackOffOnError */)
+	volumeName := v1.UniqueVolumeName("volume-name")
+	op1Name := "mount_volume"
+	operation1 := generateErrorFunc()
+	err1 := grm.Run(volumeName, "" /* operationSubName */, types.GeneratedOperations{OperationFunc: operation1, OperationName: op1Name})
+	if err1 != nil {
+		t.Fatalf("NewGoRoutine failed. Expected: <no error> Actual: <%v>", err1)
+	}
+	// Shorter than exponential backoff period, so as to trigger exponential backoff error on second
+	// operation.
+	time.Sleep(100 * time.Millisecond)
+	operation2 := generateNoopFunc()
+
+	// Act
+	err2 := grm.Run(volumeName, "" /* operationSubName */, types.GeneratedOperations{OperationFunc: operation2, OperationName: op1Name})
+
+	// Assert
+	if err2 == nil {
+		t.Fatalf("NewGoRoutine did not fail. Expected: <Failed to create operation with name \"%s\". An operation with that name already exists.> Actual: <no error>", volumeName)
+	}
+	if !exponentialbackoff.IsExponentialBackoff(err2) {
+		t.Fatalf("NewGoRoutine did not return exponential backoff, got: %v", err2)
+	}
+
+	operation3 := generateNoopFunc()
```

Comment:
the second operation returns successfully, so it won't really test the scenarios you are trying to do?

Make put another failed operation 3, and then test operation 4 with different operation name?

Reply:
It does tests the scenario I am trying to verify. Second operation does not even run because it triggers exponential backoff and `Run` function throws error and returns immediately.

Reply:
guess, what I am trying to say is - it does not matter what we do in second operation. It will *not* run and exponential backoff error is thrown and hence when third operation comes in (which is still well within 500ms of exponential backoff period) then if it did had same `operationName` it will result in exponential backoff too.

Reply:
maybe setting the second operation to return error so that we can make sure exponential backoff is not finished yet when issuing the third operation?

Reply:
okay done.